src/bench/pqs/spraylist.cpp

```diff
@@ -55,14 +55,38 @@ void
 spraylist::insert(const uint32_t &k,
                   const uint32_t &v)
 {
-    sl_add(m_q, k, READ_ADD_REM_ELASTIC_TX);
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
+}
+
+void
+spraylist::insert(const size_t &k,
+                  const size_t &v)
+{
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
 }
 
 bool
-spraylist::delete_min(uint32_t &v)
+spraylist::delete_min(size_t &k, size_t &v)
 {
-    const int ret = spray_delete_min(m_q, &v, d);
-    return (ret == 1);
+    unsigned long k_ret;
+    unsigned long v_ret;
+    int ret;
+    do {
+        ret = spray_delete_min_key(m_q, &k_ret, &v_ret, d);
```

Comment:
@schuay Yes, I think you are right, src/bench/shortest_paths.cpp calculates the correct result regardless of spurious failures. Unfortunately, the global counter is a serious scalability problem. That is why src/bench/file_shortest_paths.cpp does not really on a global counter. Instead src/bench/file_shortest_paths.cpp use a stop condition that grantee correctness as long as the priority queue has the property described in Theorem 4  in the paper  [The Contention Avoiding Concurrent Priority Queue LCPC'2016](http://www.it.uu.se/research/group/languages/software/ca_pq/relax_prio_queue_paper.pdf). I'm quite sure that KLSM satisfies this property and I don't see any reason why SprayList shouldn't.

I was surprised to see that there seems to be a small difference between using the retry loop and not using the retry loop as retries should be rare (see measurements below). Anyway, my opinion is that the proper version to measure is the one with the retry loop. I interpret @jkopinsky's answer above as that this is also his opinion. The only reason I see to expose SprayList's spurious failures to the application is to allow the application to add a custom yield or sleep before the retry. Changing "delete_min(uint32_t &v)" to look as the code that can be found below will avoid influencing the existing benchmarks. Do you want me to do that?

## SprayList Measurements
The measurements where done using the version with commit id 1ca3340a752dcf57400a9a9b67e84b2ae8bacf94. I ran it on a "AMD FX(tm)-6100 Six-Core Processor". The "retry loop version" has the retry loop and the "without retry loop" version has a delete_min method modified to look like this:

```
bool
spraylist::delete_min(uint32_t &v)
{
    size_t k_ret;
    size_t v_ret;
    const int ret = spray_delete_min_key(m_q, &k_ret, &v_ret, d);
    v = (uint32_t)v_ret;
    return ret == 1;
} 
```


### retry loop version 
```
./build/src/bench/random -p 6 spray
3890564
$ ./build/src/bench/random -p 6 spray
3893154
$ ./build/src/bench/random -p 6 spray
3902697
```
### without retry loop
```
$ ./build/src/bench/random -p 6 spray
4049361
$ ./build/src/bench/random -p 6 spray
4063919
$ ./build/src/bench/random -p 6 spray
4049860
```